src/features/todos.ts

```diff
@@ -1,8 +1,15 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 import { Todo } from '../types/Todo';
 
 export const todosSlice = createSlice({
   name: 'todos',
   initialState: [] as Todo[],
-  reducers: {},
+  reducers: {
+    setTodos: (state, action: PayloadAction<Todo[]>) => {
+      state.push(...action.payload);
```

Comment:
The 'setTodos' reducer currently pushes new todos into the existing state array. This can lead to duplicate entries if 'setTodos' is called multiple times. Consider replacing the state array with the new payload to avoid duplicates.